plugins/menu/plugin.js

```diff
@@ -188,7 +188,7 @@ CKEDITOR.plugins.add( 'menu', {
 							var item = this.editor.getMenuItem( itemName );
 
 							if ( item && ( !item.command || this.editor.getCommand( item.command ).state ) ) {
-								item.state = listenerItems[ itemName ];
+								item.state = ( item.stateFn && item.stateFn() ) || listenerItems[ itemName ];
```

Comment:
I'm not sure if a better solution won't be to just substitute listener, not add another property directly to `menu`.